From 3143cf0271e86fc1d9ba8fa3dc77087dedcd0073 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?M=C3=A5rten=20Kongstad?= <amhk@google.com>
Date: Fri, 20 Feb 2026 09:36:59 +0000
Subject: [PATCH] Fix compilation error after 37.0 finalization

Fix the following errors:

  ERROR: [...]/PinnedWindowingLayerActivity.kt:201:58 \
         Only safe (?.) or non-null asserted (!!.) calls are allowed on a nullable receiver of type 'ActivityManager.RecentTaskInfo?'
  ERROR: [...]/TaskMoveTestActivity.kt:152:30 \
         Only safe (?.) or non-null asserted (!!.) calls are allowed on a nullable receiver of type 'ActivityManager.RecentTaskInfo?'
  ERROR: [...]/AppOpsLoggingTest.kt:488:22 \
         Only safe (?.) or non-null asserted (!!.) calls are allowed on a nullable receiver of type 'BluetoothLeScanner?'.
  ERROR: [...]/AppOpsLoggingTest.kt:498:26 \
         Only safe (?.) or non-null asserted (!!.) calls are allowed on a nullable receiver of type 'BluetoothLeScanner?'.

Bug: 480974361
Test: m out/soong/.intermediates/cts/tests/framework/base/windowmanager/test-apps/app/CtsDeviceServicesTestApp/android_common/kotlin/CtsDeviceServicesTestApp.jar
Flag: NONE platform SDK finalization
Change-Id: I7b4e6e2d9251c14a8336a0b8fd0d344d52868337
---
 .../src/android/server/wm/app/PinnedWindowingLayerActivity.kt   | 2 +-
 .../app/src/android/server/wm/app/TaskMoveTestActivity.kt       | 2 +-
 .../tests/appop/src/android/app/appops/cts/AppOpsLoggingTest.kt | 2 +-
 3 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/tests/framework/base/windowmanager/test-apps/app/src/android/server/wm/app/PinnedWindowingLayerActivity.kt b/tests/framework/base/windowmanager/test-apps/app/src/android/server/wm/app/PinnedWindowingLayerActivity.kt
index e62aa115da1..75069f59e9b 100644
--- a/tests/framework/base/windowmanager/test-apps/app/src/android/server/wm/app/PinnedWindowingLayerActivity.kt
+++ b/tests/framework/base/windowmanager/test-apps/app/src/android/server/wm/app/PinnedWindowingLayerActivity.kt
@@ -198,7 +198,7 @@ class PinnedWindowingLayerActivity : Activity() {
     private fun getAppTask(): AppTask {
         val activityManager: ActivityManager = getSystemService(ActivityManager::class.java)
         val appTasks = activityManager.getAppTasks()
-        return checkNotNull(appTasks.filter { it.taskInfo.taskId == taskId }.firstOrNull()) {
+        return checkNotNull(appTasks.filter { it.taskInfo!!.taskId == taskId }.firstOrNull()) {
             "AppTask with taskId $taskId not found."
         }
     }
diff --git a/tests/framework/base/windowmanager/test-apps/app/src/android/server/wm/app/TaskMoveTestActivity.kt b/tests/framework/base/windowmanager/test-apps/app/src/android/server/wm/app/TaskMoveTestActivity.kt
index 015f3106d4f..d4351e74f3a 100644
--- a/tests/framework/base/windowmanager/test-apps/app/src/android/server/wm/app/TaskMoveTestActivity.kt
+++ b/tests/framework/base/windowmanager/test-apps/app/src/android/server/wm/app/TaskMoveTestActivity.kt
@@ -149,7 +149,7 @@ class TaskMoveTestActivity : Activity() {
         val activityManager = getSystemService(ActivityManager::class.java)
         val appTasks = activityManager.appTasks
         for (task in appTasks) {
-            if (task.taskInfo.taskId == taskId) {
+            if (task.taskInfo!!.taskId == taskId) {
                 return task
             }
         }
diff --git a/tests/tests/appop/src/android/app/appops/cts/AppOpsLoggingTest.kt b/tests/tests/appop/src/android/app/appops/cts/AppOpsLoggingTest.kt
index 9f260b39233..34ae1f3623a 100644
--- a/tests/tests/appop/src/android/app/appops/cts/AppOpsLoggingTest.kt
+++ b/tests/tests/appop/src/android/app/appops/cts/AppOpsLoggingTest.kt
@@ -485,7 +485,7 @@ class AppOpsLoggingTest {
             val btScanner = btAdapter.bluetoothLeScanner
             val scanCallback = object : ScanCallback() {}
 
-            btScanner.startScan(scanCallback)
+            btScanner!!.startScan(scanCallback)
             try {
                 eventually {
                     assertThat(asyncNoted[0].op).isEqualTo(OPSTR_BLUETOOTH_SCAN)
-- 
2.53.0.345.g96ddfc5eaa-goog

